src/rgw/support/rgw_support_bundle

```diff
@@ -164,21 +243,100 @@ class RGWSupportBundle:
+                        # select only one core if both cores exceed the size limit
+                        if sum_of_core_size > int(remaining_size[:-1]):
+                            latest_cores.pop()
+                        if not os.path.exists(dest_dump_dir):
+                            os.mkdir(dest_dump_dir)
+                    for file_name in latest_cores:
+                        shutil.copyfile(os.path.join(crash_dump_dir, file_name),
+                            os.path.join(dest_dump_dir, file_name))
+                else:
+                    shutil.copytree(crash_dump_dir, dest_dump_dir)
 
-        # add cortx components rpm version
-        cmd = "rpm -qa | grep cortx"
-        output, _, rc = SimpleProcess(cmd).run()
-        if rc == 0:
-            infile = os.path.join(RGWSupportBundle._tmp_src, 'cortx-rpms')
-            with open(infile, 'w') as fin:
-                fin.write(output.decode("utf-8"))
         RGWSupportBundle._generate_tar(bundle_id, target_path)
         RGWSupportBundle._cleanup()
 
+    @staticmethod
+    def _get_dir_size(dir_path: str):
+        """Calculate total directory size."""
+        dir_size = 0
+        for path, _, files in os.walk(dir_path):
+            for file in files:
+                file_path = os.path.join(path, file)
+                dir_size += os.path.getsize(file_path)
+        return dir_size
+
+    @staticmethod
+    def _apply_size_filter(src_dir: str, size_limit: str, file_name_reg_ex: str,
+        check_remaining_size: bool = True):
+        """Apply limit_size filter on given files."""
+        Log.info(f'Bundle {file_name_reg_ex} files.')
+        remaining_size_limit = size_limit
+        if check_remaining_size:
+            remaining_size_limit = RGWSupportBundle._get_remaining_folder_size(
+                RGWSupportBundle._tmp_src, size_limit)
+        if remaining_size_limit == '0B':
+            Log.info(f'size_limit is full for {file_name_reg_ex} files.')
+            return
+        FilterLog.limit_size(src_dir=src_dir, dest_dir=RGWSupportBundle._tmp_src,
+            size=remaining_size_limit, file_name_reg_ex=file_name_reg_ex)
+
+    @staticmethod
+    def _copy_compressed_files(src_dir: str, size_limit: str):
+        """Copy compressed log files in support bundle."""
+        for file_path in glob.glob(os.path.join(src_dir, 'rgw-*.gz')):
+            remaining_size_limit = RGWSupportBundle._get_remaining_folder_size(
+                 RGWSupportBundle._tmp_src, size_limit)
+            file_size = os.path.getsize(file_path)
+            if FilterLog._get_size_in_bytes(remaining_size_limit) < file_size:
+                Log.info('size_limit is full.')
+                break
+            shutil.move(file_path, os.path.join(
+                RGWSupportBundle._tmp_src, os.path.basename(file_path)))
+
+    @staticmethod
+    def _get_remaining_folder_size(directory: str, size_limit: str) -> str:
```

Comment:
i think similar function exists in utils also, please check once and reuse as possible

Reply:
I checked the utils support bundle and common code, but I didn't get similar function there.

Reply:
Ok i cant find it either, closing it